packages/core/src/transaction.ts

```diff
@@ -181,27 +202,32 @@ export class Transaction {
     }
   }
 
-  async begin() {
-    const queryInterface = this.sequelize.queryInterface;
-
-    if (this.sequelize.dialect.supports.settingIsolationLevelDuringTransaction) {
-      await queryInterface.startTransaction(this, this.options);
-
-      if (this.options.isolationLevel) {
-        await queryInterface.setIsolationLevel(this, this.options.isolationLevel, this.options);
-      }
+  /**
+   * Changes the isolation level of the transaction.
+   *
+   * @param isolationLevel
+   */
+  async setIsolationLevel(isolationLevel: IsolationLevel): Promise<void> {
+    await this
+      .sequelize
+      .queryInterface
+      ._setIsolationLevel(this, { ...this.options, isolationLevel });
+  }
 
-      return;
-    }
+  /**
+   * Begins a transaction
+   */
+  async #begin(): Promise<void> {
+    const queryInterface = this.sequelize.queryInterface;
 
-    if (this.options.isolationLevel) {
-      await queryInterface.setIsolationLevel(this, this.options.isolationLevel, this.options);
+    if (this.parent) {
+      return queryInterface._createSavepoint(this.parent, { ...this.options, savepointName: this.#name });
     }
 
-    await queryInterface.startTransaction(this, this.options);
+    await queryInterface._startTransaction(this, { ...this.options, transactionName: this.#name });
```

Comment:
you can't pass all options as-is. The `readOnly` option must be omitted (or set to false) if the dialect does not support the `READONLY` option on `START TRANSACTION`.

It must be omitted instead of marked as supported everywhere because of the dual role of this option (controlling both the `START TRANSACTION` query, and which database is used)

Reply:
Update now.